Common/src/drivers/CDriverBase.cpp

```diff
@@ -0,0 +1,1393 @@
+  numerics_container = new CNumerics*****[nZone]();
+  surface_movement = new CSurfaceMovement*[nZone]();
+  grid_movement = new CVolumetricMovement**[nZone]();
+  FFDBox = new CFreeFormDefBox**[nZone]();
+
+  nInst = new unsigned short[nZone];
+
+  for (iZone = 0; iZone < nZone; iZone++) {
+    nInst[iZone] = 1;
+    FFDBox[iZone] = new CFreeFormDefBox*[MAX_NUMBER_FFD];
+  }
+
+  driver_config = nullptr;
+  driver_output = nullptr;
+
+  main_config = nullptr;
+  main_geometry = nullptr;
+}
+
+unsigned short CDriverBase::GetNumberDesignVariables() const { return main_config->GetnDV(); }
+
+unsigned short CDriverBase::GetNumberFFDBoxes() const { return main_config->GetnFFDBox(); }
+
+unsigned short CDriverBase::GetNumberFFDBoxCornerPoints(unsigned short iFFDBox) const {
+  if (GetNumberFFDBoxes() == 0) {
+    SU2_MPI::Error("FFD Box definition not found in the mesh file.", CURRENT_FUNCTION);
+  }
+  if (iFFDBox >= GetNumberFFDBoxes()) {
+    SU2_MPI::Error("FFD Box index exceeds size.", CURRENT_FUNCTION);
+  }
+
+  return FFDBox[ZONE_0][iFFDBox]->GetnCornerPoints();
+}
+
+unsigned short CDriverBase::GetNumberFFDBoxControlPoints(unsigned short iFFDBox) const {
+  if (GetNumberFFDBoxes() == 0) {
+    SU2_MPI::Error("FFD Box definition not found in the mesh file.", CURRENT_FUNCTION);
+  }
+  if (iFFDBox >= GetNumberFFDBoxes()) {
+    SU2_MPI::Error("FFD Box index exceeds size.", CURRENT_FUNCTION);
+  }
+
+  return FFDBox[ZONE_0][iFFDBox]->GetnControlPoints();
+}
+
+unsigned long CDriverBase::GetNumberFFDBoxSurfacePoints(unsigned short iFFDBox) const {
+  if (GetNumberFFDBoxes() == 0) {
+    SU2_MPI::Error("FFD Box definition not found in the mesh file.", CURRENT_FUNCTION);
+  }
+  if (iFFDBox >= GetNumberFFDBoxes()) {
+    SU2_MPI::Error("FFD Box index exceeds size.", CURRENT_FUNCTION);
+  }
+
+  return FFDBox[ZONE_0][iFFDBox]->GetnSurfacePoint();
+}
+
+vector<unsigned short> CDriverBase::GetFFDBoxMarkerIDs(unsigned short iFFDBox) const {
+  if (GetNumberFFDBoxes() == 0) {
+    SU2_MPI::Error("FFD Box definition not found in the mesh file.", CURRENT_FUNCTION);
+  }
+  if (iFFDBox >= GetNumberFFDBoxes()) {
+    SU2_MPI::Error("FFD Box index exceeds size.", CURRENT_FUNCTION);
+  }
+
+  vector<unsigned short> values;
+
+  for (auto iPoint = 0ul; iPoint < GetNumberFFDBoxSurfacePoints(iFFDBox); iPoint++) {
+    values.push_back(FFDBox[ZONE_0][iFFDBox]->Get_MarkerIndex(iPoint));
+  }
+  return values;
+}
+
+vector<unsigned long> CDriverBase::GetFFDBoxVertexIDs(unsigned short iFFDBox) const {
+  if (GetNumberFFDBoxes() == 0) {
+    SU2_MPI::Error("FFD Box definition not found in the mesh file.", CURRENT_FUNCTION);
+  }
+  if (iFFDBox >= GetNumberFFDBoxes()) {
+    SU2_MPI::Error("FFD Box index exceeds size.", CURRENT_FUNCTION);
+  }
+
+  vector<unsigned long> values;
+
+  for (auto iPoint = 0ul; iPoint < GetNumberFFDBoxSurfacePoints(iFFDBox); iPoint++) {
+    values.push_back(FFDBox[ZONE_0][iFFDBox]->Get_VertexIndex(iPoint));
+  }
+  return values;
+}
```

Comment:
Making the driver the interface to the FFD does not feel like the best idea, because you need to basically duplicate the FFD interface here.
Would it not be possible to wrap the FFD with swig? And then in python do `driver.GetFFD(iFFD).GetNumberSurfacePoints()` and so on?
Can you stash this for a different PR?

Reply:
Sure, we can discuss more streamlined ways to expose the FFD functionality to Python in the future. I will remove the FFD functions from the current PR and move it as part of PR #1750

Reply:
FFD functions removed in commit [3c8f5cd](https://github.com/su2code/SU2/pull/1300/commits/3c8f5cd84eea3a870c6f3b0bf2b74fdbdfead498)